client/blocks/get-apps/desktop-download-card.jsx

```diff
@@ -53,32 +61,62 @@ class DesktopDownloadCard extends Component {
 		}
 	}
 
+	getPlatformImage( platform ) {
+		switch ( platform ) {
+			case 'MacIntel':
+			case MAC_LINK:
+				return <SVGIcon name="apple-logo" size="24" icon={ Apple } />;
+			case 'Linux i686':
+			case 'Linux i686 on x86_64':
+			case LINUX_TAR_LINK:
+			case LINUX_DEB_LINK:
+				return <SVGIcon name="linux-logo" size="24" icon={ Linux } />;
+			case WINDOWS_LINK:
+			default:
+				return <SVGIcon name="windows-logo" size="24" icon={ Windows } />;
+		}
+	}
+
 	getTranslateComponents( platform ) {
 		switch ( platform ) {
 			case 'MacIntel':
 				return {
 					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
 					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
 					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+					...( displayJetpackAppBranding && {
+						firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
+						secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
+						thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
+					} ),
 				};
```

Comment:
Nit: rather than mutating every property in object, WDYT about returning a different object? IMO it may be little easier to comprehend and, eventually, remove the conditional. If we choose to make this modification, we should apply it to the other object spreads in this function as well. 

```suggestion
				return displayJetpackAppBranding ? {
          firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
          secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
          thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
        } : {
					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
				};
```

Reply:
I appreciate your perspective here. Zooming out, I can see how the mutation would be hard to parse. I've gone ahead to create separate objects in https://github.com/Automattic/wp-calypso/pull/69674/commits/59edabb39eac9a83792917b4e052316b30742863, as per your suggestion.